starter-ui/src/main/java/org/eclipse/starter/ui/StarterServlet.java

```diff
@@ -1,70 +1,70 @@
+
 import java.io.File;
 import java.io.IOException;
 import java.lang.invoke.MethodHandles;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Comparator;
+import java.util.function.Function;
 import java.util.logging.Logger;
 import java.util.stream.Stream;
 import java.util.zip.ZipOutputStream;
-import org.eclipse.starter.mavengenerator.CliMavenContext;
-import org.eclipse.starter.mavengenerator.EeArchetypeGenerateParameters;
-import org.eclipse.starter.mavengenerator.ZipCodec;
 
 @WebServlet(urlPatterns = {"/download.zip"}, name = "StarterServlet")
 public class StarterServlet extends HttpServlet {
 
+
+    private static final Function<String, Boolean> IS_EMPTY = (String s) -> s == null || s.isEmpty();
+
     private static final Logger logger = Logger.getLogger(
             MethodHandles.lookup().lookupClass().getName());
-    private static final String DOWNLOADABLE_FILE_NAME = "jakartaee-cafe.zip";
+    private static final String DOWNLOADABLE_FILE_NAME = "jakartaee-project.zip";
+
 
-    private static class Parameters {
-        String archetypeGroupId = null;
-        String archetypeArtifactId = null;
-        String archetypeVersion = null;
-        String groupId = null;
-        String artifactId = null;
-        String version = null;
-        String profile = null;
-    }
-    
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
-            throws ServletException, IOException {
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp)
+            throws IOException {
         logger.info("Generating project from archetype");
 
-        Parameters p = parametersFromrequest(req);
+        final Parameters parametersFromrequest = this.parametersFromrequest(req);
 
         logger.fine(() -> String.format("Archetype properties: archetypeGroupId=%s, archetypeArtifactId=%s, "
-                + "archetypeVersion=%s, profile=%s, groupId=%s, artifactId=%s, version=%s", 
-                p.archetypeGroupId, p.archetypeArtifactId, p.archetypeVersion, p.profile,
-                p.groupId, p.artifactId, p.version));
-
-        prepareResponse(resp, DOWNLOADABLE_FILE_NAME);
-        Path generatedDirectory = generateProjectFromArchetype(p);
-        try {
-            Path projectDirectory = Files.list(generatedDirectory).findAny().get();
-            zipDirectoryToOutput(projectDirectory, resp.getOutputStream());
+                        + "archetypeVersion=%s, profile=%s, groupId=%s, artifactId=%s, version=%s",
+                parametersFromrequest.archetypeGroupId, parametersFromrequest.archetypeArtifactId,
+                parametersFromrequest.archetypeVersion, parametersFromrequest.profile,
+                parametersFromrequest.groupId, parametersFromrequest.artifactId, parametersFromrequest.version));
+
+        final var downloadableFileName = IS_EMPTY.apply(parametersFromrequest.artifactId)
```

Comment:
This is interesting code. I would use a static method instead of a lambda. But I don't mind, if you like it this way :)

Reply:
made it more KISS :-)